File: packages/whisker-hello-element/src/ios/HelloView.swift
// Plain Lynx UI subclass for the `Hello` element. No annotation;
// instantiated by Lynx via the behavior `HelloModule`'s
// `definition()` registers (see `HelloModule.swift`).
//
// `@objc(HelloView)` pins the Obj-C class name to the bare
// `HelloView` (instead of `<SwiftPM-target>.HelloView`) so the
// codegen plugin's `NSClassFromString` lookup can find it under
// either name.

import UIKit
import WhiskerModuleApi

@objc(HelloView)
public final class HelloView: WhiskerUI<UIView> {
    @objc public override func createView() -> UIView {
        let v = UIView()
        v.backgroundColor = .systemPink
        return v
    }
}

File: packages/whisker-hello-element/src/android/HelloView.kt
// Plain Lynx UI subclass for the `Hello` element. No annotation;
// instantiated by the Lynx behavior `HelloModule`'s `definition()`
// registers (see `HelloModule.kt`). The single-arg `(WhiskerContext)`
// constructor matches the convention the KSP L-2c registration code
// expects.

package rs.whisker.elements.hello

import android.content.Context
import android.graphics.Color
import android.view.View
import rs.whisker.runtime.WhiskerContext
import rs.whisker.runtime.WhiskerUI

open class HelloView(context: WhiskerContext) : WhiskerUI<View>(context) {
    override fun createView(context: Context): View {
        val view = View(context)
        view.setBackgroundColor(Color.YELLOW)
        return view
    }
}
